Ship_Game/AI/StrategyAI/FleetRatios.cs

```diff
@@ -48,23 +48,23 @@ public FleetRatios(Empire empire)
             MinCarriers    = 0;
             MinSupport     = 0;
             MinCombatFleet = 0;
-            SetFleetRatios();
+            SetFleetRatios(empire.Universe.P.UseRandomRatios);
 
 
         }
 
-        public void SetFleetRatios()
+        public void SetFleetRatios(bool useRandomFleetSizes)
         {
             // fighters, corvettes, frigate, cruisers, capitals, troopShip,bombers,carriers,support
 
             int[] counts;
 
-            if      (OwnerEmpire.CanBuildCapitals)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCapitals);
-            else if (OwnerEmpire.CanBuildBattleships) counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildBattleships);
-            else if (OwnerEmpire.CanBuildCruisers)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCruisers);
-            else if (OwnerEmpire.CanBuildFrigates)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildFrigates);
-            else if (OwnerEmpire.CanBuildCorvettes)   counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCorvettes);
-            else                                      counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildFighters);
+            if      (OwnerEmpire.CanBuildCapitals)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCapitals, IsRandomizedAIFleetSizes);
+            else if (OwnerEmpire.CanBuildBattleships) counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildBattleships, IsRandomizedAIFleetSizes);
+            else if (OwnerEmpire.CanBuildCruisers)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCruisers, IsRandomizedAIFleetSizes);
+            else if (OwnerEmpire.CanBuildFrigates)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildFrigates, IsRandomizedAIFleetSizes);
+            else if (OwnerEmpire.CanBuildCorvettes)   counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCorvettes, IsRandomizedAIFleetSizes);
+            else                                      counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildFighters, IsRandomizedAIFleetSizes);
```

Comment:
```suggestion
        public void SetFleetRatios(bool useRandomRatios)
        {
            // fighters, corvettes, frigate, cruisers, capitals, troopShip,bombers,carriers,support

            Range[] counts;

            if      (OwnerEmpire.CanBuildCapitals)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCapitals);
            else if (OwnerEmpire.CanBuildBattleships) counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildBattleships);
            else if (OwnerEmpire.CanBuildCruisers)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCruisers);
            else if (OwnerEmpire.CanBuildFrigates)    counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildFrigates);
            else if (OwnerEmpire.CanBuildCorvettes)   counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildCorvettes);
            else                                      counts = ResourceManager.GetFleetRatios(BuildRatio.CanBuildFighters);

            SetCounts(counts, useRandomRatios);
            CountIndexed = counts;
        }
```

So you just return the range from the static function and there are no loops involved. 
In SetCounts you can use either min or Generate depends on the flag